crypto/keys/src/traits.rs

```diff
@@ -48,60 +80,88 @@ pub trait DistinguishedEncoding: Sized {
     /// Create a new object from the given DER-encoded SubjectPublicKeyInfo.
     fn try_from_der(src: &[u8]) -> Result<Self, KeyError>;
 
+    /// Create the standardized DER-encoded representation of this object and
+    /// write to the provided buffer.
+    fn to_der_slice<'a>(&self, buff: &'a mut [u8]) -> &'a [u8];
+
     /// Create the standardized DER-encoded representation of this object.
-    fn to_der(&self) -> Vec<u8>;
+    #[cfg(feature = "alloc")]
+    fn to_der(&self) -> Vec<u8> {
+        let mut buff = alloc::vec![0; <Self as DistinguishedEncoding>::der_size()];
+        self.to_der_slice(&mut buff);
+        buff
+    }
 }
 
-/// A trait indicating that a string fingerprint can be generated for an
-/// object.
+/// Maximum length of the DER buffer required for
+/// [`DistinguishedEncoding::to_der()`]
+pub const DER_MAX_LEN: usize = 128;
```

Comment:
:wrench: Suggest moving this inside of the `DistinguishedEncoding` trait to provide better locality

Reply:
i believe associated consts make traits non object safe, and wouldn't work with the static assertions elsewhere

Reply:
Thanks for that explanation, I didn't know about object safety and actually a little disappointed consts prevent it